cfme/physical/provider/redfish.py

```diff
@@ -0,0 +1,69 @@
+import attr
+
+from widgetastic_patternfly import BootstrapSelect, Input
+from wrapanapi.systems import RedfishSystem
+
+from cfme.common.provider import DefaultEndpoint, DefaultEndpointForm
+from . import PhysicalProvider
+
+
+class RedfishEndpoint(DefaultEndpoint):
+    api_port = 443
+    security_protocol = 'SSL'
+
+    @property
+    def view_value_mapping(self):
+        return {
+            'security_protocol': self.security_protocol,
+            'hostname': self.hostname,
+            'api_port': self.api_port
+        }
+
+
+class RedfishEndpointForm(DefaultEndpointForm):
+    security_protocol = BootstrapSelect('default_security_protocol')
+    api_port = Input('default_api_port')
+
+
+@attr.s(hash=False)
+class RedfishProvider(PhysicalProvider):
+    STATS_TO_MATCH = ['num_server']
+    type_name = 'redfish'
+    endpoints_form = RedfishEndpointForm
+    string_name = 'Physical Infrastructure'
+    mgmt_class = RedfishSystem
+    refresh_text = "Refresh Relationships and Power States"
+    db_types = ["Redfish::PhysicalInfraManager"]
+    settings_key = 'ems_redfish'
+    log_name = 'redfish'
+
+    @property
+    def mgmt(self):
+        from cfme.utils.providers import get_mgmt
+        endpoint = self.data['endpoints']['default']
+        d = {
```

Comment:
I'm not sure why this is required, and why the default processing of the yaml in `cfme.utils.providers.get_mgmt` isn't sufficient here. The attributes you're referencing would have been parsed from the same yaml, and `get_mgmt` uses the default endpoint.

Not necessarily saying its wrong, but I'd be interested to know where `get_mgmt` is not serving its purpose here, and document with in-line comments what the intent of passing specific data here is.

Reply:
Additionally there's none of the protections offered in the implementation provided by BaseProvider.mgmt property.